Tutorial: Update cookieless tracking (#8713)
PostHog/posthog.com: contents/tutorials/cookieless-tracking.md

---
title: How to do cookieless tracking with PostHog
sidebar: Docs
showTitle: true
featuredImage: >-
  https://res.cloudinary.com/dmukukwp6/image/upload/posthog.com/contents/images/cookieless-tracking.png
featuredVideo: 'https://www.youtube-nocookie.com/embed/3V3fbz6sgPk'
featuredTutorial: false
date: 2024-06-14
author:
  - joe-martin
tags:
  - configuration
  - product os
---

Normally, PostHog stores some information about the user in their browser using a cookie. This approach is typical for analytics tools and enables user tracking across sessions, caching feature flag data, and more. 

There are some situations where you don't want to use cookies and do cookieless tracking instead. These include when:

- You have concerns about user privacy or regulation such as [GDPR](/docs/integrate/gdpr) or [HIPAA](/docs/privacy/hipaa-compliance).

- You have your own system for identifying users across multiple sessions, rely on server-side tracking, or don’t need to track user identities at all.

- You hate cookie banners.

> This tutorial shows how to configure PostHog's JavaScript Web SDK to do cookieless tracking by using page memory to store user data.

Reviewer: ```suggestion
This tutorial shows how to configure PostHog's [JavaScript Web SDK](/docs/libraries/js) to do cookieless tracking by using page memory to store user data.
```

> **Note:** When using cookies, PostHog stores data as a first-party cookie. We don't track users across different sites (like largely blocked [third-party cookies](https://en.wikipedia.org/wiki/Third-party_cookies) do). It also means the same cookie works across subdomains like `posthog.com` and `eu.posthog.com`.

<GDPRForm />

## Step 1: Decide where to store the data

It is helpful first to know what data is being stored and why. Specifically, PostHog stores the following information in the user’s browser:

- User `distinct_id`
- Session ID
- Device ID
- Active [feature flags](/docs/user-guides/feature-flags)
- [Super properties](/docs/integrate/client/js#super-properties)
- Configuration options (e.g., whether [session recording](/docs/user-guides/recordings) is enabled)

> If you want to use PostHog without cookies, you must store some of this data elsewhere. Although PostHog has [multiple persistence options](/docs/libraries/js#persistence), the most straightforward is to store it in page memory. This is done by setting `persistance` to `memory`.

Reviewer: Setting persistance to memory confused me. My immediate questions were "how and where do I do this". Only later I saw its covered in the next step

```suggestion
If you want to use PostHog without cookies, you must store some of this data elsewhere. Although PostHog has [multiple persistence options](/docs/libraries/js#persistence), the most straightforward is to store it in page memory. We show you how to do this in the next step.
```
